src/peft/tuners/lora/layer.py

```diff
@@ -217,37 +216,34 @@ def single_batched_forward(self, x: torch.Tensor) -> torch.Tensor:
         result = result.to(previous_dtype)
         return result
 
-    def multi_batched_forward(self, x: torch.Tensor) -> torch.Tensor:
-        num_adapters = len(self.id_to_adapter_dict)
+    def multi_batched_forward(self, x: torch.Tensor, adapter_indices: torch.Tensor) -> torch.Tensor:
+        final_output = self._linear(x)
+        if self.disable_adapters:
+            return final_output
+
+        id_to_adapter_dict = dict(enumerate(self.lora_A, start=1))  # 0 is base
```

Comment:
creating on the fly for each call and on each module is redundant. Better to pass this as `kwarg`